### PR TITLE
Add network to docker how-to

### PR DESCRIPTION
**Pull request by padraic-shafer**
Updated example of Multi-host scalable Docker How-To:
* Uses a custom bridge network to connect tiled and database services
* Using `host` or `default` bridge network for Docker containers is brittle

We might consider adding a docker-compose file to simplify this example.

### CODE REVIEW

**Review by dylanmcreynolds — APPROVED** [review]
Great addition. I left a couple items comments in but I approve.

File: docs/source/how-to/docker.md [hunk 1]
```diff
@@ -130,6 +144,7 @@ export TILED_DATABASE_PASSWORD=db_secret
 mkdir postgres-data
 docker run -d \
   --name tiled-test-postgres \
+  --net=tilednet \
```
[the remark this answers is not in the document]
Comment (padraic-shafer) [reply docs/source/how-to/docker.md:147]: Right you are.  That's a few lines above this. :)

File: docs/source/how-to/docker.md [hunk 2]
```diff
@@ -122,6 +122,20 @@ docker run \
...
+To connect these services we create a bridge network.
+
+```
+docker network create -d bridge tilednet
+```
+
+```{note}
+We chose to use a custom bridge network here for two reasons:
+1. The host network is not supported for Docker Desktop for Mac/Windows
+2. The default bridge network does not resolve container DNS entries by hostname
```
[the remark this answers is not in the document]
Comment (dylanmcreynolds) [reply docs/source/how-to/docker.md:136]: This is interesting. Thanks. That could explain somethings that colleagues saw on mac. Thanks!

**Comment by padraic-shafer** [comment]
Based on the approval and discussion, I'll go ahead and merge this docs update.

**Pull request merged**